spread/project_test.go

```diff
@@ -78,19 +78,46 @@ suites:
  tests/:
   summary: mock tests
 `)
-	tmpdir := c.MkDir()
-	err := ioutil.WriteFile(filepath.Join(tmpdir, "spread.yaml"), spreadYaml, 0644)
-	c.Assert(err, IsNil)
-	err = os.MkdirAll(filepath.Join(tmpdir, "tests"), 0755)
-	c.Assert(err, IsNil)
+	for _, tc := range []struct {
+		filename string // the file name to write the project to
+		subdir   string // the project subdir
+		errMsg   string // the expected error message
+	}{
+		{"spread.yaml", "", ""},
+		{".spread.yaml", "", ""},
+		{"other.yaml", "", "cannot load project file from .*: cannot find spread.yaml or .spread.yaml"},
+		{"spread.yaml", "subdir1", "cannot list suite tests/: open .*: no such file or directory"},
+		{"spread.yaml", "subdir2", ""},
+		{"spread.yaml", ".", ""},
+		{"spread.yaml", "/", ""},
+	} {
```

Comment:
The code above wasn't very nice to read due to the wrong indentation, and it's now getting worse due to the way the for loop is embedding a struct and a list in it.  Can we please reorganize things for it to look clean?  The table should be moved out of the test (typical condensed syntax we use), with a named struct, tests defined in the table using field names as usual, ("reroot" instead of "subdir" maybe?), etc.

Not sure if the function is already here, but we have a nice "reindent" function in Chisel to allow the embedded yaml to be properly indented with the rest of the code.  Let me know if you need a link.